plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeTranslator.java

```diff
@@ -40,6 +43,20 @@ private OrcTypeTranslator() {}
             }
             return Optional.of(new LongTimestampToVarcharCoercer(timestampType, varcharType));
         }
+        if (isVarcharType(fromOrcType.getOrcTypeKind()) && toTrinoType instanceof TimestampType timestampType) {
+            VarcharType varcharType = fromOrcType.getLength()
+                    .map(VarcharType::createVarcharType)
+                    .orElseGet(VarcharType::createUnboundedVarcharType);
```

Comment:
`.orElse(VARCHAR);`

Reply:
it looks like we could always use `VARCHAR` here.
i don't think `fromOrcType.getLength()` matters.

Reply:
What if the schema of a column changes from unbounded varchar to a bounded varchar and them to timestamp, won't it affect the result ?

Reply:
I don't have a test coverage for this -  but when we try to append data to an existing ORC file - the footer could capture the new schema

Reply:
i thought this is being used on read, not on write?

also, i think orc files are immutable, you cannot append to an existing data file

Reply:
But I read somewhere for insert-only type of tables, it just appends to an existing file for new set of data and with an updated footer.

Reply:
insert-only tables mean that you can insert to them, but you cannot update, delete from them
this is a hive concept

yes, i vaguely remember some mention of some streaming-ingest for ORC where files are built incrementally, but Trino does not do that and will never do that. Object storage doesn't support that either.

Reply:
Okay !! We can assume that - varchar length specified in the footer will be in always greater than or equal to the length of Slice in case of VariableWidthBlock i.e we won't require a truncating of varchar during reads.

Reply:
Yes !! So we will remove that commit.